src/content/capture/canvas-filter/js/main.js

```diff
@@ -0,0 +1,51 @@
+/*
+*  Copyright (c) 2016 The WebRTC project authors. All Rights Reserved.
+*
+*  Use of this source code is governed by a BSD-style license
+*  that can be found in the LICENSE file in the root of the source
+*  tree.
+*/
+
+/* global main */
+
+'use strict';
+
+const source = document.querySelector('#source');
+const canvasIn = document.querySelector('#canvas-source');
```

Comment:
For demonstration purposes, this is fine.
For performance reasons, it would be better to have intermediate canvas-buffers to not be in the DOM (created inline with document.createElement('canvas') or be using OffscreenCanvas.

Reply:
Good point. OffscreenCanvas is probably the best way. I had them visible for debugging.

Reply:
However, I'll do that in the next iteration. Added a TODO.